test/test_ui_find.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2025 João Sousa
+# Copyright (C) 2025 Francisco Lisboa
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+from test.picardtestcase import PicardTestCase
+
+from picard.metadata import Metadata
+from picard.tags import preserved_tag_names
+
+from picard.ui.find import FindBox
+from picard.ui.itemviews import MainPanel
+from picard.ui.itemviews.basetreeview import BaseTreeView
+
+
+class FindBoxTest(PicardTestCase):
+    """Test the FindBox widget functionality"""
+
+    def test_tag_names_available(self):
+        """Test that tag_names() returns valid tags for filtering"""
+        tags = set(preserved_tag_names())
+        self.assertIsInstance(tags, set)
+        self.assertGreater(len(tags), 0)
+        self.assertIn('artist', tags)
+        self.assertIn('album', tags)
+        self.assertIn('title', tags)
+
+    def test_findbox_class_exists(self):
+        """Test that FindBox class can be imported and has required attributes"""
+        self.assertTrue(hasattr(FindBox, 'findChanged'))
+        self.assertTrue(hasattr(FindBox, '_query_changed'))
+        self.assertTrue(hasattr(FindBox, 'clear'))
+        self.assertTrue(hasattr(FindBox, 'set_focus'))
+
+    def test_filter_button_text_logic(self):
+        """Test the logic for updating filter button text from FindBox._show_filter_dialog"""
+        test_cases = [
+            ([], "Filters"),
+            (["filename"], "filename"),
+            (["filename", "artist"], "filename, artist"),
```

Comment:
With the recent change in https://github.com/metabrainz/picard/pull/2668/commits/95c2e94056c4286c162efe6b6878b8f4eabfdfad this should be changed to `(['filename', 'artist'], "2 filters"),`.

Reply:
Also, a really minor point but in general we use single quotes for keys and such.  See https://github.com/metabrainz/picard/blob/master/CONTRIBUTING.md#strings-quoting-single-or-double-quotes for more information.